depthai_bridge/src/DisparityConverter.cpp

```diff
@@ -19,6 +19,28 @@ DisparityConverter::DisparityConverter(
     _rosBaseTime = ::ros::Time::now();
 }
 
+void DisparityConverter::updateRosBaseTime()
+{
+    ::ros::Time currentRosTime = ::ros::Time::now();
+    std::chrono::time_point<std::chrono::steady_clock> currentSteadyTime =
+        std::chrono::steady_clock::now();
+    // In nanoseconds
+    auto expectedOffset = std::chrono::duration_cast<std::chrono::nanoseconds>(currentSteadyTime - _steadyBaseTime).count();
+    uint64_t previousBaseTimeNs = _rosBaseTime.toNSec();
+    _rosBaseTime = _rosBaseTime.fromNSec(currentRosTime.toNSec() - expectedOffset);
```

Comment:
If I understand correctly here, this is constantly shifting the `_rosBaseTime` (time that this instance was created), based on the change in the steady clock. This is because the steady clock is monotonically increasing, but under some circumstances the ROS time may jump forwards or backwards. So we're compensating for that here?